### PR TITLE
Add crossplane kubernetes provider

### PR DESCRIPTION
**Pull request by hmariset**
Setup crossplane provider to be deployed to respective EaaS clusters.

Jira-Url: https://issues.redhat.com/browse/KFLUXVNGD-89

### CODE REVIEW

File: config/k8s/provider-deployment-runtime-config.yaml [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+    deploymentTemplate:
+      spec:
+        selector: {}
+        template:
+          spec:
+            containers:
+              - name: package-runtime
+                args:
+                  - '--sanitize-secrets'
+                  - '--debug'
```
Comment (amisstea) [inline config/k8s/provider-deployment-runtime-config.yaml:16]: We probably shouldn't enable debug by default. Let's omit it for now.

File: config/base/provider.yaml [hunk 2]
```diff
@@ -0,0 +1,7 @@
+---
+apiVersion: pkg.crossplane.io/v1
+kind: Provider
+metadata:
```
Comment (amisstea) [inline config/base/provider.yaml:4]: We need the `argocd.argoproj.io/sync-options: SkipDryRunOnMissingResource=true` annotation set on this resource so ArgoCD can successfully sync the required CRD first.

File: config/k8s/provider-deployment-runtime-config.yaml [hunk 3]
```diff
@@ -0,0 +1,16 @@
+---
+  apiVersion: pkg.crossplane.io/v1beta1
+  kind: DeploymentRuntimeConfig
+  metadata:
```
Comment (amisstea) [inline config/k8s/provider-deployment-runtime-config.yaml:4]: We need the `argocd.argoproj.io/sync-options: SkipDryRunOnMissingResource=true` annotation here too.

File: config/k8s/provider-deployment-runtime-config.yaml (hunk 1 above)
Comment (hmariset) [reply config/k8s/provider-deployment-runtime-config.yaml]: Updated.

File: config/base/provider.yaml (hunk 2 above)
Comment (hmariset) [reply config/base/provider.yaml:4]: Added the annotation.

File: config/k8s/provider-deployment-runtime-config.yaml (hunk 3 above)
Comment (hmariset) [reply config/k8s/provider-deployment-runtime-config.yaml]: Added the annotation.

File: config/k8s/provider-deployment-runtime-config.yaml
Comment (amisstea) [inline config/k8s/provider-deployment-runtime-config.yaml:1]: The patches on the Provider are currently redundant. To simplify things, I think we should move this file under `config/base`, add `spec.runtimeConfigRef` directly to the `Provider` and drop the patches.

Comment (hmariset) [reply config/k8s/provider-deployment-runtime-config.yaml:1]: Updated

**Review by amisstea — APPROVED** [review]

**Pull request merged**